src/Microsoft.AspNetCore.Http.Abstractions/Routing/RouteValueDictionary.cs

```diff
@@ -167,17 +167,17 @@ public RouteValueDictionary(object values)
 
             set
             {
-                if (string.IsNullOrEmpty(key))
+                if (key == null)
```

Comment:
This was a bug. We have no reason to disallow the empty string as a key, and this is the only place we enforced this.

Reply:
Test for this (I didn't see a test in the PR)

Reply:
We don't usually test null checks 🤷‍♂️

Reply:
I mean test that an empty string as the key works to avoid a future regression.

Reply:
added